source/python/idetect/nlp_models/relevance/relevance.py

```diff
@@ -0,0 +1,205 @@
+
+from idetect.model import Relevance
+
+class RelevanceModel(object):
+    def __init__(self, model_path=None):
+        self.model = self.load_model(model_path=model_path)
+
+    def load_model(self, model_path=None):
+        if model_path and os.path.isfile(model_path):
+            clf = joblib.load(model_path)
+        else:
+            default_model_path = 'relevance_classifier.pkl'
+            if os.path.isfile(default_model_path):
+                clf = joblib.load(default_model_path)
+            else:
+                url = 'https://s3-us-west-2.amazonaws.com/idmc-idetect/relevance_models/relevance_classifier.pkl'
+                r = requests.get(url, stream=True)
+                if not os.path.isfile(default_model_path):
+                    try:
+                        os.makedirs(os.path.dirname(default_model_path))
+                    except OSError as exc: # Guard against race condition
+                        if exc.errno != errno.EEXIST:
+                            raise
+                with open(default_model_path, 'wb') as f:
+                    for chunk in r.iter_content(chunk_size=1024):
+                        if chunk:  # filter out keep-alive new chunks
+                            f.write(chunk)
+                clf = joblib.load(default_model_path)
+        return clf
+
+    def predict(self, text):
+        try:
+            relevance = self.model.transform(pd.Series(text))[0]
+        except:
+            # if error occurs, classify as most likely category
+            relevance = 'yes'
+
+        if relevance == 'yes':
+            return Relevance.DISPLACEMENT
+        else:
+            return Relevance.NOT_DISPLACEMENT
+
+class Tokenizer(TransformerMixin):
+    def __init__(self, stop_words=None):
+        self.stop_words = stop_words
+        self.tokenizer = WordPunctTokenizer()
+        
+    def prepare_tokens(self, text, stop_words):
+        tokens = self.tokenizer.tokenize(text)
+        tokens = [t for t in tokens if len(t) > 2]
+        tokens = [t for t in tokens if t not in stop_words]
+        tokens = [t.lower() for t in tokens]
+        tokens = [t for t in tokens if not t.isdigit()]
+        return tokens
+    
+    def fit(self, X, *args):
+        return self
+    
+    def transform(self, X, *args):
+        X = X.map(lambda x: self.prepare_tokens(x, self.stop_words))
+        return X
+
+class Stemmer(TransformerMixin):
+    def __init__(self, stop_words):
+        self.stop_words = stop_words
+        self.tokenizer = WordPunctTokenizer()
+        self.stemmer = PorterStemmer()
+        
+    def prepare_stems(self, text, stop_words):
+        tokens = self.tokenizer.tokenize(text)
+        tokens = [t for t in tokens if len(t) > 2]
+        tokens = [t for t in tokens if t not in stop_words] 
+        stems = [self.stemmer.stem(t) for t in tokens]
+        stems = [s.lower() for s in stems]
+        stems = [s for s in stems if not s.isdigit()]
```

Comment:
Does `self.stemmer.stem(t)` return digits for tokens that weren't `isdigit()` to begin with? Maybe "1000s" => "1000"?